src/Discord.Net.Commands/Dependencies/DependencyMap.cs

```diff
@@ -48,6 +50,8 @@ public bool TryAddTransient<T>() where T : class, new()
             var t = typeof(T);
             if (map.ContainsKey(t))
                 return false;
+            if (typeof(T) == typeof(IDependencyMap) || typeof(T) == typeof(CommandService))
+                throw new InvalidOperationException("The dependency map cannot contain services directly added as IDependencyMap or CommandService. Only Implimentations and Derivatives are permitted");
```

Comment:
I'd probably just keep it simple and explain why. `IDependencyMap is used internally and cannot be added as a dependency`.